### PR TITLE
S1 FLL-25 Feat intergrate Gemini AI

### PR DESCRIPTION
**Pull request by tuongnhatnguyenvan**
I have completed integrating Gemini Ai into the project and using markdown !
@khoanguyen-st @quyenduc26 @ABon485 @ToLinh2004 @luantranthannh

### CODE REVIEW

**Review by quyenduc26 — APPROVED** [review]

File: app/page.tsx [hunk 1]
```diff
@@ -1,48 +1,106 @@
...
 
-const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL || 'http://localhost:4000';
+const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL || "";
 
-export default function ChatInterface() {
-  const [bot, setBot] = useState<BotInfoType | null>(null);
-  const [messages, setMessages] = useState<{ role: 'user' | 'assistant'; content: string }[]>([]);
-  const [input, setInput] = useState('');
+// AI Service
+const useAIService = () => {
```
Comment (khoanguyen-st) [inline app/page.tsx]: Blocking: Move custom hooks into a folder like `src/hooks` for easily managing

File: app/page.tsx [hunk 2]
```diff
@@ -1,48 +1,106 @@
...
+      };
+
+      const result = await model.generateContent({
+        contents: [{ role: "user", parts: [{ text: userMessage }] }],
+        generationConfig,
+      });
+
+      return result.response.text();
+    } catch (error) {
+      console.error("Error interacting with Gemini AI:", error);
```
Comment (khoanguyen-st) [inline app/page.tsx]: Blocking: Remove log

File: app/page.tsx [hunk 3]
```diff
@@ -1,48 +1,106 @@
...
+      return "Sorry, I have some error. Please try again.";
```
Comment (khoanguyen-st) [inline app/page.tsx]: Suggestion: Should use constant here

File: app/page.tsx [hunk 4]
```diff
@@ -1,48 +1,106 @@
...
+    }
+  };
+
+  return { handleAIResponse };
+};
 
+// Bot Service
+const useBotService = () => {
```
Comment (khoanguyen-st) [inline app/page.tsx]: Blocking: Same as above

File: app/page.tsx [hunk 5]
```diff
@@ -1,48 +1,106 @@
...
+    setChatHistory((prev) => [
+      ...prev,
+      { role: "user", content: inputMessage },
+    ]);
+
+    const aiResponse = await handleAIResponse(inputMessage);
+    setChatHistory((prev) => [
+      ...prev,
+      { role: "assistant", content: aiResponse },
+    ]);
```
Comment (khoanguyen-st) [inline app/page.tsx]: Suggestion: Should refactor this code when handle the Chat History

File: app/page.tsx (hunk 2 above)
Comment (tuongnhatnguyenvan) [reply app/page.tsx:35]: @khoanguyen-st , dạ anh ơi, cái này mình bắt lỗi thì nên log ra lỗi chứ a hè, em ko bk tại sao lại ko log lỗi ra á

**Comment by tuongnhatnguyenvan** [comment]
@khoanguyen-st 
I have finished the edit base on your feedback!

**Review by ToLinh2004 — APPROVED** [review]

**Review by khoanguyen-st — APPROVED** [review]

**Pull request merged**